extensions-core/druid-catalog/src/main/java/org/apache/druid/catalog/http/MoveColumn.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.catalog.http;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nullable;
+
+import java.util.Objects;
+
+/**
+ * JSON payload for the reorder column API.
+ */
+public class MoveColumn extends TableEditRequest
+{
+  public enum Position
+  {
+    FIRST,
+    LAST,
+    BEFORE,
+    AFTER
+  }
+
+  @JsonProperty
+  public final String column;
+  @JsonProperty
+  public final Position where;
+  @Nullable
+  @JsonProperty
+  public final String anchor;
+
+  @JsonCreator
+  public MoveColumn(
+      @JsonProperty("column") final String column,
+      @JsonProperty("where") final Position where,
```

Comment:
Should we fail construction of this class if anchor is null, and position is one of BEFORE or AFTER?

Reply:
I see we are doing the validation in the TableEditor class. I think its best to move the validation as high up in the stack as possible, and then users can assume that the object has already been validated. What do you think?